distributed/client.py

```diff
@@ -3575,6 +3591,68 @@ def get_events(self, topic: str = None):
         """
         return self.sync(self.scheduler.events, topic=topic)
 
+    async def _handle_events(self, sub, handler):
+        while self.status == "running":
+            event = await sub.get()
+            ret = handler(event)
+            if inspect.isawaitable(ret):
+                await ret
+
+    def subscribe_topic(self, topic, handler):
+        """Subscribe to a topic and execute a handler for every received event
+
+        Parameters
+        ----------
+        topic: str
+            The topic name
+        handler: callable or coroutinefunction
+            A handler called for every received event. The handler must accept a
+            single argument `event` which is a tuple `(timestamp, msg)` where
+            timestamp refers to the clock on the scheduler.
+
+        Example
+        -------
+
+        >>> import logging
+        >>> logger = logging.getLogger("myLogger")  # Log config not shown
+        >>> client.subscribe_topic("topic-name", lambda: logger.info)
+
+        See Also
+        --------
+        dask.distributed.Client.unsubscribe_topic
+        dask.distributed.Client.get_events
+        dask.distributed.Client.log_event
+        """
+        if topic not in self.event_subscriptions:
+            sub = self.event_subscriptions[topic] = Sub(topic, client=self)
+        else:
+            sub = self.event_subscriptions[topic]
+
+        if topic in self.event_handlers:
+            fut = self.event_handlers[topic]
+            fut.cancel()
+
+        self.event_handlers[topic] = fut = asyncio.ensure_future(
+            self._handle_events(sub=sub, handler=handler)
+        )
+
+    def unsubscribe_topic(self, topic):
+        """Unsubscribe from a topic and remove event handler
+
+        See Also
+        --------
+        dask.distributed.Client.subscribe_topic
+        dask.distributed.Client.get_events
+        dask.distributed.Client.log_event
+        """
+        if topic in self.event_handlers:
+            fut = self.event_handlers.pop(topic)
+            fut.cancel()
+            sub = self.event_subscriptions.pop(topic)
+            sub.stop()
+        else:
```

Comment:
This is mostly convenience stuff to execute an event handler on client side. The actual subscription is handled via the extension